[HTTP Binding]: adding metadata options for client auth during mTLS (#3050)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-bindings/http.md

  metadata:
  - name: url
    value: http://something.com
  - name: MTLSRootCA
    value: /Users/somepath/root.pem #<path to root CA> or <pem encoded string>
  - name: MTLSClientCert
    value: /Users/somepath/client.pem #<path to client cert> or <pem encoded string>
  - name: MTLSClientKey
    value: /Users/somepath/client.key #<path to client key> or <pem encoded string>

```

## Spec metadata fields

| Field              | Required | Binding support | Details | Example |
|--------------------|:--------:|--------|--------|---------|
| url                | Y        | Output |The base URL of the HTTP endpoint to invoke | `http://host:port/path`, `http://myservice:8000/customers`

> | MTLSRootCA         | N        | Output |Path to root ca certificate or pem encoded string |

Reviewer: Can you provide examples for the paths. Always good to clarify if this can be absolute path or relative path and if relative, relative to what.

{{< /tabs >}}

## Using mTLS or enabling client TLS authentication along with HTTPS

> The HTTP binding can also be configured to use mTLS or client TLS authentication along with HTTPS by providing the `MTLSRootCA`, `MTLSClientCert` and `MTLSClientKey` metadata fields in the binding component.

Reviewer 2: ```suggestion
You can configure the HTTP binding to use mTLS or client TLS authentication along with HTTPS by providing the `MTLSRootCA`, `MTLSClientCert`, and `MTLSClientKey` metadata fields in the binding component.
```

> These fields can be passed as a file path or as a pem encoded strings. If the file path is provided, the file will be read and the contents will be used. If the pem encoded string is provided, the string will be used as is.

Reviewer 2: ```suggestion
These fields can be passed as a file path or as a pem encoded string. 
- If the file path is provided, the file will be read and the contents will be used. 
- If the pem encoded string is provided, the string will be used as is.
```

> When these fields are configured, dapr sidecar will use the provided certificate to authenticate itself with the server during the TLS handshake process.

Reviewer 2: ```suggestion
When these fields are configured, the Dapr sidecar will use the provided certificate to authenticate itself with the server during the TLS handshake process.
```

### When to use:

> This can be used when the server with which the HTTP binding is configured to communicate with, requires mTLS or client TLS authentication.

Reviewer 2: ```suggestion
You can use this when the server with which the HTTP binding is configured to communicate requires mTLS or client TLS authentication.
```

{{< /tabs >}}

## Using mTLS or enabling client TLS authentication along with HTTPS
You can configure the HTTP binding to use mTLS or client TLS authentication along with HTTPS by providing the `MTLSRootCA`, `MTLSClientCert`, and `MTLSClientKey` metadata fields in the binding component.

These fields can be passed as a file path or as a pem encoded string. 

> - If the file path is provided, the file will be read and the contents will be used. 

Reviewer: ```suggestion
- If the file path is provided, the file is read and the contents are used. 
```

> - If the pem encoded string is provided, the string will be used as is.

Reviewer: ```suggestion
- If the pem encoded string is provided, the string is used as is.
```

> When these fields are configured, the Dapr sidecar will use the provided certificate to authenticate itself with the server during the TLS handshake process.

Reviewer: ```suggestion
When these fields are configured, the Dapr sidecar uses the provided certificate to authenticate itself with the server during the TLS handshake process.
```
